cmd/bbolt/main.go

```diff
@@ -1394,20 +1394,43 @@ func (cmd *benchCommand) runReadsSequentialNested(db *bolt.DB, options *BenchOpt
 	})
 }
 
-func checkProgress(results *BenchResults, finishChan chan interface{}, stderr io.Writer) {
-	ticker := time.Tick(time.Second)
+// StartProgress starts a goroutine to report progress periodically.
+func (r *BenchResults) StartProgress(w io.Writer) *BenchResults {
+	r.stop = make(chan struct{})
+	r.report = make(chan struct{})
+	go r.checkProgress(w)
+	return r
+}
+
+// StopProgress stops the goroutine that reports progress periodically.
+func (r *BenchResults) StopProgress() {
+	close(r.stop)
+}
+
+// ReportProgress immediately without waiting for the next tick.
+func (r *BenchResults) ReportProgress() {
+	select {
+	case r.report <- struct{}{}:
+	default:
+	}
+}
+
+func (r *BenchResults) checkProgress(w io.Writer) {
 	lastCompleted, lastTime := int64(0), time.Now()
 	for {
+		var t time.Time
 		select {
-		case <-finishChan:
+		case <-r.stop:
 			return
-		case t := <-ticker:
-			completed, taken := results.CompletedOps(), t.Sub(lastTime)
-			fmt.Fprintf(stderr, "Completed %d requests, %d/s \n",
-				completed, ((completed-lastCompleted)*int64(time.Second))/int64(taken),
-			)
-			lastCompleted, lastTime = completed, t
+		case <-r.report:
+			t = time.Now()
+		case t = <-time.After(time.Second):
 		}
+		completed, taken := r.CompletedOps(), t.Sub(lastTime)
+		fmt.Fprintf(w, "Completed %d requests, %d/s \n",
```

Comment:
I don't know if you want to track this with a log file on every line individually, but for the sake of console spam, maybe add a CR?

Reply:
Sorry @tjungblu . I couldn't understand your comment. The aim of this CR is to enable reporting progress on console while the benchmark is running. Can you elaborate a bit more on this?

Reply:
perfect, then why print a line for each instead of using a carriage return to have the progress report in a single line?